File: Floater.java
import java.awt.Color;
import java.awt.Font;
import java.awt.Graphics;

public class Floater {
  static Color TextColor;
  
  static Font TextFont;
  
  int Frames;
  
  int XMin;
  
  int XMax;
  
  int YSpeed;
  
  int xpos;
  
  int ypos;
  
  int XSpeed;
  
  int Direction;
  
  String TextString;
  
  boolean done;
  
  public Floater(int paramInt1, int paramInt2, String paramString, int paramInt3, int paramInt4, int paramInt5, int paramInt6) {
    this.Frames = paramInt6;
    this.TextString = paramString;
    this.XMin = paramInt3;
    this.XMax = paramInt4;
    this.YSpeed = paramInt5;
    this.XSpeed = 1;
    this.Direction = 1;
    this.xpos = paramInt1;
    this.ypos = paramInt2;
    this.done = false;
  }
  
  static void Init(Color paramColor, Font paramFont) {
    TextColor = paramColor;
    TextFont = paramFont;
  }
  
  public void Paint(Graphics paramGraphics) {
    paramGraphics.setColor(TextColor);
    paramGraphics.setFont(TextFont);
    paramGraphics.drawString(this.TextString, this.xpos, this.ypos);
  }
  
  public void Update() {
    this.ypos += this.YSpeed;
    this.xpos += this.XSpeed;
    if (this.Frames % 3 == 0) {
      this.XSpeed += this.Direction;
      if (this.XSpeed == this.XMax) {
        this.Direction = -1;
      } else if (this.XSpeed == this.XMin) {
        this.Direction = 1;
      } 
    } 
    this.Frames--;
    if (this.Frames < 0)
      this.done = true; 
  }
}


/* Location:              C:\Users\brody\Downloads\invader.jar!\Floater.class
 * Java compiler version: 1 (45.3)
 * JD-Core Version:       1.1.3
 */

File: web/js/Floater.js
// Floater.js — port of Floater.java
// Floating score/damage text that drifts upward.

let textColor = '#ffffff';
let textFont  = '12px monospace';

export function initFloater(color, font) {
  textColor = color;
  textFont  = font;
}

export class Floater {
  constructor(x, y, text, xMin, xMax, ySpeed, frames) {
    this.xpos       = x;
    this.ypos       = y;
    this.textString = text;
    this.xMin       = xMin;
    this.xMax       = xMax;
    this.ySpeed     = ySpeed;
    this.xSpeed     = 1;
    this.direction  = 1;
    this.frames     = frames;
    this.done       = false;
  }

  paint(ctx) {
    ctx.fillStyle = textColor;
    ctx.font      = textFont;
    ctx.fillText(this.textString, this.xpos, this.ypos);
  }

  update() {
    this.ypos += this.ySpeed;
    this.xpos += this.xSpeed;
    if (this.frames % 3 === 0) {
      this.xSpeed += this.direction;
      if (this.xSpeed === this.xMax)       this.direction = -1;
      else if (this.xSpeed === this.xMin)  this.direction =  1;
    }
    this.frames--;
    if (this.frames < 0) this.done = true;
  }
}
